course6/toyapp-adam/math/matrix.cc

```diff
@@ -0,0 +1,135 @@
+#include "matrix.h"
+
+#include <cmath>
+
+namespace Math {
+
+namespace {
+	constexpr auto PI = std::atan(1.0) * 4.0;
+}
+
+Matrix44 Matrix44::Zero()
+{
+	Matrix44 m;
+
+	for(unsigned i = 0; i < 16; ++i)
+		m.mMatrix[i] = 0.0f;
+
+	return m;
+}
+
+Matrix44 Matrix44::Identity()
+{
+	Matrix44 m = Zero();
+
+	m.mMatrix[ 0] = 1.0f;
+	m.mMatrix[ 5] = 1.0f;
+	m.mMatrix[10] = 1.0f;
+	m.mMatrix[15] = 1.0f;
+
+	return m;
+}
+
+Matrix44 Matrix44::Translation(Vector3D position)
+{
+	Matrix44 m = Identity();
+
+	m.mMatrix[TRANSLATE_X] = position.x;
+	m.mMatrix[TRANSLATE_Y] = position.y;
+	m.mMatrix[TRANSLATE_Z] = position.z;
+
+	return m;
+}
+
+Matrix44 Matrix44::Rotation(Vector3D yawPitchRoll)
+{
+	Matrix44 m = Zero();
+
+	const float ch = std::cos(yawPitchRoll.x);
+	const float sh = std::sin(yawPitchRoll.x);
+	const float cp = std::cos(yawPitchRoll.y);
+	const float sp = std::sin(yawPitchRoll.y);
+	const float cb = std::cos(yawPitchRoll.z);
+	const float sb = std::sin(yawPitchRoll.z);
+
+	const float shsp = sh * sp;
+	const float chsp = ch * sp;
```

Comment:
1. You don't need to make them const
2. It's OK to precalculate them for readability, but you need to be aware that it's not required from the performance point of view.

Reply:
I'm aware it's not required, maybe it's a bad habit of mine. However I do like to make stuff const by default (not function arguments though), but yeah maybe for simple things I shouldn't.